spec/lib/pdf_filler/nj1040_pdf_spec.rb

```diff
@@ -0,0 +1,201 @@
+            expect(pdf_fields[name_field]).to eq "Hopper Grace B"
+          end
+        end
+
+        context "with suffix" do
+          let(:submission) {
+            create :efile_submission, tax_return: nil, data_source: create(
+              :state_file_nj_intake,
+              primary_first_name: "Grace",
+              primary_last_name: "Hopper",
+              primary_suffix: "JR"
+            ) }
+          it 'fills pdf with LastName FirstName Suf' do
+            expect(pdf_fields[name_field]).to eq "Hopper Grace JR"
+          end
+        end
+
+        context "with suffix and middle initial" do
+          let(:submission) { 
+            create :efile_submission, tax_return: nil, data_source: create(
+            :state_file_nj_intake, 
+            primary_first_name: "Grace", 
+            primary_last_name: "Hopper", 
+            primary_middle_initial: "B", 
+            primary_suffix: "JR" 
+          ) }
+          it 'fills pdf with LastName FirstName MI Suf' do
+            expect(pdf_fields[name_field]).to eq "Hopper Grace B JR"
+          end
+        end
+      end
+
+      context "joint filer" do
+        context "same last name" do
+          let(:submission) { 
+            create :efile_submission, tax_return: nil, data_source: create(
+            :state_file_nj_intake, 
+            primary_first_name: "Bert", 
+            primary_last_name: "Muppet", 
+            primary_middle_initial: "S",
+            spouse_first_name: "Ernie",
+            spouse_last_name: "Muppet",
+            spouse_ssn: "123456789"
+            ) }
+          it 'fills pdf with LastName FirstName & FirstName' do
+            expect(pdf_fields[name_field]).to eq "Muppet Bert S & Ernie"
+          end
+        end
+
+        context "different last names" do
+          let(:submission) {
+            create :efile_submission, tax_return: nil, data_source: create(
+              :state_file_nj_intake,
+              primary_first_name: "Blake",
+              primary_last_name: "Lively",
+              primary_middle_initial: "E",
+              spouse_first_name: "Ryan",
+              spouse_last_name: "Reynolds",
+              spouse_ssn: "123456789"
+            ) }
+          it 'fills pdf with LastName FirstName & LastName FirstName' do
+            expect(pdf_fields[name_field]).to eq "Lively Blake E & Reynolds Ryan"
+          end
+        end
+      end
+    end
+
+    describe "address fields" do
+      let(:submission) {
+        create :efile_submission, tax_return: nil, data_source: create(
+          :state_file_nj_intake,
+        ) }
+
+      it 'enters values into PDF' do
+        # address values from zeus_one_dep.xml
+        expect(pdf_fields["SpousesCU Partners SSN if filing jointly"]).to eq "391 US-206 B"
+        expect(pdf_fields["CountyMunicipality Code See Table page 50"]).to eq "Hammonton"
+        expect(pdf_fields["State"]).to eq "NJ"
+        expect(pdf_fields["ZIP Code"]).to eq "08037"
```

Comment:
I couldn't figure out how to pass these into the factory and have them actually be used without being overwritten by the DF raw data in the factory... so at the end of the day this is just referencing the values in that XML instead of setting them in the test context (like the other tests in this file do). I'd love to fix this but I'm not sure how.

Reply:
see comment on factory!